src/test/java/book_store/controller/BookControllerTest.java

```diff
@@ -0,0 +1,233 @@
+        }
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get all books")
+    void getAll_ReturnsAllBooks_ExpectedSuccess() throws Exception {
+        List<BookDto> expected = new ArrayList<>();
+        expected.add(new BookDto().setId(1L).setTitle("Kobzar").setAuthor("Shevchenko")
+                .setPrice(BigDecimal.valueOf(10.99)).setIsbn("9460303332081"));
+        expected.add(new BookDto().setId(2L).setTitle("Avatar").setAuthor("Unknow")
+                .setPrice(BigDecimal.valueOf(20.99)).setIsbn("9460306342021"));
+        expected.add(new BookDto().setId(3L).setTitle("Terminator").setAuthor("Arnold")
+                .setPrice(BigDecimal.valueOf(30.99)).setIsbn("9460301332081"));
+
+        MvcResult result = mockMvc.perform(
+                        get("/books")
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper.readValue(result.getResponse().getContentAsByteArray(), BookDto[].class);
+        Assertions.assertEquals(3, actual.length);
+        for (BookDto book : expected) {
+            Assertions.assertTrue(Arrays.stream(actual)
+                    .anyMatch(a -> a.getId().equals(book.getId())
+                            && a.getTitle().equals(book.getTitle())
+                            && a.getAuthor().equals(book.getAuthor())
+                            && a.getPrice().equals(book.getPrice())
+                            && a.getIsbn().equals(book.getIsbn())));
+        }
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get book by ID")
+    void getBookById_ReturnsBookById_ExpectedSuccess() throws Exception {
+        BookDto expectedBook = new BookDto()
+                .setId(1L)
+                .setTitle("Kobzar")
+                .setAuthor("Shevchenko")
+                .setPrice(BigDecimal.valueOf(10.99))
+                .setIsbn("9460303332081");
+
+        MvcResult result = mockMvc.perform(
+                        get("/books/{id}", 1L)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actualBook = objectMapper.readValue(result.getResponse().getContentAsByteArray(), BookDto.class);
+
+        Assertions.assertEquals(expectedBook.getId(), actualBook.getId());
+        Assertions.assertEquals(expectedBook.getTitle(), actualBook.getTitle());
+        Assertions.assertEquals(expectedBook.getAuthor(), actualBook.getAuthor());
+        Assertions.assertEquals(expectedBook.getPrice(), actualBook.getPrice());
+        Assertions.assertEquals(expectedBook.getIsbn(), actualBook.getIsbn());
```

Comment:
EqualsBuilder or implementing equals in model will allow to get rid of each field comparison